auth/src/main/java/org/aerogear/mobile/security/SecurityCheckExecutor.java

```diff
@@ -10,107 +10,161 @@
              */
             public T withSecurityCheck(@NonNull final SecurityCheck check) {
-                checks.add(check);
+                checks.add(nonNull(check, "check"));
                 return (T) this;
             }
 
             /**
              * Adds a new security check.
-             * @param checkType type of security check to be added
-             * @return this
+             *
+             * @param checkType type of security check to be added.
+             * @return {@link T} this.
+             * @throws IllegalArgumentException if checkType is null.
              */
             public T withSecurityCheck(@NonNull final SecurityCheckType checkType) {
-                checks.add(checkType.getSecurityCheck());
+                checks.add(nonNull(checkType, "checkType").getSecurityCheck());
                 return (T) this;
             }
 
             /**
              * Sets the metric singleThreadService to be used.
              *
-             * @param metricsService the metric singleThreadService to be used
-             * @return this
+             * @param metricsService the metric singleThreadService to be used.
+             * @return this.
              */
             public T withMetricsService(@Nullable final MetricsService metricsService) {
                 this.metricsService = metricsService;
                 return (T) this;
             }
 
+            /**
+             * Gets the context for the device.
+             *
+             * @return {@link Context}.
+             */
             protected Context getCtx() {
                 return ctx;
             }
 
+            /**
+             * Gets the metric service being used.
+             *
+             * @return {@link MetricsService}.
+             */
             protected MetricsService getMetricsService() {
                 return metricsService;
             }
 
+            /**
+             * Gets all the checks that are to be tested.
+             *
+             * @return {@link Collection<SecurityCheck>}.
+             */
             protected Collection<SecurityCheck> getChecks() {
                 return checks;
             }
 
             /**
              * Builds the executor according to the passed in parameters.
              *
-             * @return the executor instance
+             * @return the executor instance.
              */
             public abstract K build();
         }
 
+        /**
+         * Creates a Builder for AsyncSecurityCheckExecutor.
+         *
+         * @param ctx {@link Context} for the device.
+         * @return {@link AsyncSecurityCheckExecutor.Builder}.
+         * @throws IllegalArgumentException if ctx is null.
+         */
         private AsyncSecurityCheckExecutor.Builder newAsyncBuilder(@NonNull final Context ctx) {
-            return new AsyncSecurityCheckExecutor.Builder(ctx);
+            return new AsyncSecurityCheckExecutor.Builder(nonNull(ctx, "context"));
```

Comment:
the constructor itself already has the check, so I don't think it's required to do the `nonNull` check here.

Reply:
Ya you're right, I'll remove it here